subsystem/cluster_test.go

```diff
@@ -4010,3 +4010,346 @@ var _ = Describe("disk encryption", func() {
 		Expect(*h.StatusInfo).Should(ContainSubstring("The host's TPM version is not supported"))
 	})
 })
+
+var _ = Describe("Ovirt provider tests", func() {
+	var (
+		ctx         = context.Background()
+		clusterID   strfmt.UUID
+		cluster     *models.Cluster
+		clusterCIDR = "10.128.0.0/14"
+		serviceCIDR = "172.30.0.0/16"
+	)
+
+	BeforeEach(func() {
+		registerClusterReply, err := userBMClient.Installer.RegisterCluster(ctx, &installer.RegisterClusterParams{
+			NewClusterParams: &models.ClusterCreateParams{
+				Name:             swag.String("test-cluster"),
+				BaseDNSDomain:    "example.com",
+				OpenshiftVersion: swag.String(openshiftVersion),
+				PullSecret:       swag.String(pullSecret),
+				ClusterNetworks:  []*models.ClusterNetwork{{Cidr: models.Subnet(clusterCIDR), HostPrefix: 23}},
+				ServiceNetworks:  []*models.ServiceNetwork{{Cidr: models.Subnet(serviceCIDR)}},
+				SSHPublicKey:     sshPublicKey,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		cluster = registerClusterReply.GetPayload()
+		clusterID = *cluster.ID
+		generateClusterISO(clusterID, models.ImageTypeMinimalIso)
+	})
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	It("install cluster with oVirt hosts - only masters", func() {
+		By("set hosts with hw info and setup essential host steps")
+		inventoryOvirtInfo := getoVirtInventory()
+		ips := hostutil.GenerateIPv4Addresses(3, defaultCIDRv4)
+		h1 := registerNode(ctx, clusterID, "h1", ips[0])
```

Comment:
registerNode already contains generateEssentialHostStepsWithInventory (default one). I think that you could probably use registerHost